### ISSUE TITLE
AV issue

### ISSUE THREAD

**Issue by RaskaSlaanesh** [intiface/intiface-game-haptics-router#36]
AV bitdefender labels one of the files during installation as malware, more specifially 
**Trojan.GenericKD.38640302**
and its file **IntifaceGameHapticsRouter\unins000.dat**
its weird, but probably a problem, that should be looked into.

**Comment by rbairwell** [comment]
Windows defender has just alerted (after download)  of "PUA:Win32/Puwaders.C!ml" (Potentially Unwanted Software) in intiface-game-haptics-router-win-x64-v18-installer.exe->(inno#000025) -and VirusTotal says[ 20 antivirus](https://www.virustotal.com/gui/file/c5cfa50072b6a7f70068dbebe57586be15ad2be94675f5075f1f04f8e7fc570c/detection) systems flagged it - mainly due to [SharpMonoInjector.dll](https://www.virustotal.com/gui/file/c5cfa50072b6a7f70068dbebe57586be15ad2be94675f5075f1f04f8e7fc570c/relations) (which I'm guessing is used to inject the haptics routing into the game and hence modifies 3rd party software and hence causes the antivirus warnings).

Honestly, I'm not sure how this could be worked around bearing in mind it needs to be able to intercept and alter calls to gamepads to send them to Intiface...

**Comment by blackspherefollower** [comment]
Duplicate of #11 ?

**Comment by VaporMoon** [comment]
Windows immediately quarantined the v18 installer on hitting download as it claims to have detected **Trojan:Win32/Tnega!MSR**.
The v17 installer downloaded and installed with no warnings. Built v18 myself and it's fine too. Intiface Central also gives no errors.

**Comment by SkYScouter** [comment]
Windows defender is also not allowing me to download v18 of the GHR. Downloaded interface central just fine

**Comment by qdot** [comment]
I'm gonna start working on this today. May release a v19 that just kills SharpMonoInjector entirely, at the cost of meaning we won't be able to filter processes, but eh. I'm gonna be rebuilding the whole GHR anyways so it's a decent stop gap.

**Comment by SkYScouter** [comment]
thank you @qdot for the reply, I'll keep a look out for v19! Also, does v17 of the interface work with the latest version of the Intiface central?

**Comment by qdot** [comment]
@SkYScouter Yup it should, only new feature for v18 was UWP hooking, so it just works with more games.

**Comment by qdot** [comment]
@SkYScouter @VaporMoon Ok, v19 is up. The *only* change is that I built locally and signed using my new cert, so I can't really confirm that this won't end up tripping antivirus again, but we'll see. 

https://github.com/intiface/intiface-game-haptics-router/releases/tag/v19

**Comment by VaporMoon** [comment]
@qdot v19 download and installed without tripping anything.

**Comment by qdot** [comment]
It usually takes a day or two for windows defender to start getting mad, but I already through the file through VirusTotal and it only got a couple of DllInject detections (which are valid), so we'll see what happens. Gonna leave this open for like a week longer, will close after that if I stop hearing reports.

This may also be the last version of the GHR in this form, I'm working on a new system that's a little more extensible. More info on that once it's slightly more baked though, it's not even in the oven yet. :3

**Comment by SkYScouter** [comment]
@qdot, yep didn't trip at all for me, was having a little trouble setting up Elden ring with v17 so I'll try this. thx

**Comment by moukimyr** [comment]
Unfortunately, I just had v19 get flagged by Windows Defender, but I'm assuming it's fine to allow it anyways?

**Comment by qdot** [comment]
@moukimyr Yup. Not surprising that it happened. I just need to completely remove our process enumeration filtering code. :/

**Comment by qdot** [comment]
This is still happening, and unfortunately getting rid of the process filter is a bit more work than I thought, haven't had time to cover it yet.

**Comment by Aonodensetsu** [comment]
v19 is flagged by 42/72 on virustotal
